app/resources/admin.html.template

```diff
@@ -138,35 +170,81 @@
+    selected_language_col.appendChild(selected_language);
+
+    var title_col = row.insertCell(cur_col++);
+    var title = document.createElement('textarea');
+    title.cols = 72;
+    title.id = table_id + '_language_value_' + iteration;
+    title_col.appendChild(title);
+    return iteration; 
+  }
+
+  function add_another_language() {
+    add_language();
+    for(var id_num = 1; id_num < CUSTOM_MESSAGE_TABLE_IDS.length; id_num++){
+      add_language_span(CUSTOM_MESSAGE_TABLE_IDS[id_num], 'not selected');
+    }
+  }
+
   /* Hides a language from the user's view and marks it so that it will not be
    * submitted.
    */
   function remove_language(row_number) {
-    var table = $('language_table');
-    var row = table.rows[row_number];
-    row.hidden = true;
-    $('language_selection_' + row_number).value = 'unused';
+    for (var id_num = 0; id_num < CUSTOM_MESSAGE_TABLE_IDS.length; id_num++) {
+      var table_name = CUSTOM_MESSAGE_TABLE_IDS[id_num];
+      var table = $(table_name + '_table');
+      var row = table.rows[row_number];
+      row.hidden = true;
+      if (table_name == 'repo_titles') {
+        $(table_name + '_language_selection_' + row_number).value = 'unused';
+      } else {
+        document.getElementsByClassName('language_selected_' + row_number).value
+                                       = 'unused';
+      }
+    }
   }
 
   // Adds rows to the languages table for each language already in the database
   function add_initial_languages() {
     // Saved languages and titles for this repository
+    var all_view_config_json = {{all_view_config_json|safe|default:'[]'}};
     var LANGUAGES = {{view_config_json.language_menu_options|safe|default:'[]'}};
-    var TITLES = {{view_config_json.repo_titles|safe|default:'[]'}};
 
     //foreach language, add that language and its title
-    for (var lang_num = 0; lang_num < LANGUAGES.length; lang_num++) {
-      var row_number = add_language();
-      var language = LANGUAGES[lang_num];
-      $('language_selection_' + row_number).value = language;
-      if (TITLES[language] != undefined) {
-        $('language_title_' + row_number).value = TITLES[language];
+    for (var id_num = 0; id_num < CUSTOM_MESSAGE_TABLE_IDS.length; id_num++) {
+      var table_id = CUSTOM_MESSAGE_TABLE_IDS[id_num];
+      var table_name = all_view_config_json[table_id];
+      for (var lang_num = 0; lang_num < LANGUAGES.length; lang_num++) {
+        var language = LANGUAGES[lang_num];
+        if (table_id == 'repo_titles') {
+          var row_number = add_language();
+          $(table_id + '_language_selection_' + row_number).value = language;
+        } else {
+          var row_number = add_language_span(table_id, language);
+          update_language_selected(row_number);
+        }
+        if (table_name[language] != undefined) {
+          $(table_id + '_language_value_' + row_number).value
+          = table_name[language];
```

Comment:
new line should indent by 4 spaces like:

$(table_id + '_language_value_' + row_number).value
    = table_name[language];